Mathlib/GroupTheory/Coxeter/Basic.lean

```diff
@@ -56,162 +49,119 @@ presentation corresponding to a Coxeter matrix which is registered in a Coxeter
 
 ## TODO
 
-* The canonical map from the type to the Coxeter group `W` is an injection.
-* A group `W` registered in a Coxeter system is a Coxeter group.
-* A Coxeter group is an instance of `IsCoxeterGroup`.
+* The simple reflections of a Coxeter system are distinct.
 
 ## Tags
 
 coxeter system, coxeter group
+
 -/
 
+open Function
+
+/-! ### Coxeter groups -/
 
-universe u
+namespace CoxeterMatrix
 
-noncomputable section
+variable {B B' : Type*} (M : CoxeterMatrix B) (e : B ≃ B')
 
-variable {B : Type*} [DecidableEq B]
-variable (M : Matrix B B ℕ)
+/-- The Coxeter relation associated to a Coxeter matrix $M$ and two indices $i, i' \in B$.
+That is, the relation $(s_i s_{i'})^{M_{i, i'}}$, considered as an element of the free group
+on $\{s_i\}_{i \in B}$.
+If $M_{i, i'} = 0$, then this is the identity, indicating that there is no relation between
+$s_i$ and $s_{i'}$. -/
+def relation (i i' : B) : FreeGroup B := (FreeGroup.of i * FreeGroup.of i') ^ M i i'
 
-namespace CoxeterGroup
+/-- The set of all Coxeter relations associated to the Coxeter matrix $M$. -/
+def relationsSet : Set (FreeGroup B) := Set.range <| uncurry M.relation
 
-namespace Relations
+/-- The Coxeter group associated to a Coxeter matrix $M$; that is, the group
+$$\langle \{s_i\}_{i \in B} \vert \{(s_i s_{i'})^{M_{i, i'}}\}_{i, i' \in B} \rangle.$$ -/
+def group := PresentedGroup M.relationsSet
```

Comment:
Please add a type annotation here.
```suggestion
def group : sorry := PresentedGroup M.relationsSet
```
If this is a `Type _`, then it should be renamed to Group.

Reply:
thanks; I also made it protected